src/liboslexec/batched_llvm_gen.cpp

```diff
@@ -5933,7 +5933,7 @@ LLVMGEN (llvm_gen_gettextureinfo)
     FuncSpec func_spec("get_textureinfo_uniform");
     const char * funcName = rop.build_name(func_spec);
 
-    bool requires_binning = (!Filename.is_uniform()) | (!Dataname.is_uniform()) | use_coords;
+    bool requires_binning = (!Filename.is_uniform()) || (!Dataname.is_uniform()) || use_coords;
```

Comment:
These were all changed to bitwise from logical to avoid generating branching code.  As this is during compilation, probably not measurable impact.  But if we disable the check with "-Wno-bitwise-instead-of-logical" then these files would not even need to be in this pull request.

Reply:
I put the guards (i.e. kept the bitwise ops) in all code paths that are called during shader execution or that are in code we are trying to vectorize.

There were a few instances -- including this one, which is where we're generating IR -- where the code is not trying to be vectorized, not called by the shaders themselves, and not especially performance sensitive. And in those circumstances, I reverted to the logical operations because that's the real meaning of this statement and there is nothing to gain from eliminating a single branch here.

I also looked at godbolt and see that whether `&&` and `||` are turned into branchless '&` and `|` automatically by the C++ optimizer depends heavily on the specific compiler and version. Modern icc and clang are pretty good at it in many circumstances. gcc is not. So I think the use of bitwise ops is only important in areas that are extremely performance sensitive and we don't want to take any chances that the compiler will fail to optimize away the branch.

Reply:
Agreed that the change for batched_llvm_gen.cpp and other places isn't performance critical.  And that change is fine.  Just noting as they were all bools it is somewhat equivalent when the compiler does automatically turn them into & and |, kinda shows that they are valid to be & and | to begin with.

Also a compiler "might" turn && and || are turned into branchless & and | automatically.  Turning optimization off though probably won't turn && and || are turned into branchless & and |.  So one might prefer to never generate the branch in the first place, to better ensure similar execution during debug.  Also the example I posted, clearly a jmp is still generated with the || at O3, because it is contractually obligated to by language rules around short-circuit.  And Clang might be good and that automatic conversion if it can prove no side effects (which is good feature), but other compilers or optimization levels may not be and still generate jmps.

Reply:
I've updated the PR to revert all the individual line-by-line pragmas in favor of just disabling this warning for all of liboslexec.